diff -ur a/src/.depend b/src/.depend
--- a/src/.depend	2009-07-03 17:14:48.000000000 +0200
+++ b/src/.depend	2009-07-03 17:15:32.000000000 +0200
@@ -318,4 +318,17 @@
 f90_SM_clones.cmo: targets.cmi omega.cmi models.cmi fusion.cmi 
 f90_SM_clones.cmx: targets.cmx omega.cmx models.cmx fusion.cmx 
 count.cmo: topology.cmi thoList.cmi combinatorics.cmi 
-count.cmx: topology.cmx thoList.cmx combinatorics.cmx 
+count.cmx: topology.cmx thoList.cmx combinatorics.cmx
+
+# FeynRules preliminary
+$(FR_MODULES:.ml=.cmi) : model.cmi coupling.cmi
+
+$(FR_MODULES:.ml=.cmo) : thoList.cmi rCS.cmi product.cmi options.cmi model.cmi \
+    coupling.cmi combinatorics.cmi color.cmi $(FR_MODULES:.ml=.cmi)
+
+$(FR_MODULES:.ml=.cmx) : thoList.cmx rCS.cmx product.cmx options.cmx model.cmi \
+    coupling.cmi combinatorics.cmx color.cmx $(FR_MODULES:.ml=.cmi)
+
+$(FR_BINARIES:.ml=.cmo) : targets.cmo omega.cmo $(FR_MODULES:.ml=.cmo) fusion.cmo colorize.cmo
+
+$(FR_BINARIES:.ml=.cmx) : targets.cmx omega.cmx $(FR_MODULES:.ml=.cmx) fusion.cmx colorize.cmx
diff -ur a/src/Makefile.src b/src/Makefile.src
--- a/src/Makefile.src	2009-07-03 17:14:48.000000000 +0200
+++ b/src/Makefile.src	2009-07-03 17:17:28.000000000 +0200
@@ -38,6 +38,10 @@
 MODEL_MODULES = 
 endif
 
+# Feynrules preliminary
+FR_MODULES = $(shell find . -name 'fr*.ml')
+FR_BINARIES = $(shell find . -name 'f90_fr*.ml')
+
 MODULES = \
 	pmap.ml thoList.ml thoArray.ml thoString.ml rCS.ml \
 	cache.ml trie.ml lapack.ml linalg.ml \
@@ -51,7 +55,8 @@
 	oVM.ml whizard.ml \
 	comphep_syntax.ml comphep_lexer.mll comphep_parser.mly comphep.ml \
 	$(MODEL_MODULES) colorize.ml \
-	omega_syntax.ml omega_lexer.mll omega_parser.mly omega.ml
+	omega_syntax.ml omega_lexer.mll omega_parser.mly omega.ml \
+	$(FR_MODULES)
 
 GUI_MODULES = \
 	thoGButton.ml thoGWindow.ml thoGMenu.ml thoGDraw.ml
@@ -80,7 +85,8 @@
 	f90_Simplest f90_Simplest_univ \
 	f90_Xdim f90_GravTest \
 	f90_SM_km f90_UED f90_Zprime \
-	f90_Template
+	f90_Template \
+	$(FR_BINARIES:.ml=)
  
 # Colorized for WHiZard
 PROGRAMS_released += \
diff --recursive -u a/src/colorize.ml b/src/colorize.ml
--- a/src/colorize.ml	2009-08-04 17:28:10.000000000 +0200
+++ b/src/colorize.ml	2009-08-04 17:28:22.000000000 +0200
@@ -102,6 +102,8 @@
       | Prop_Majorana -> Prop_Col_Majorana   (* Spin 1/2 octets. *)
       | Prop_Feynman -> Prop_Col_Feynman   (* Spin 1 states, massless. *)
       | Prop_Unitarity -> Prop_Col_Unitarity   (* Spin 1 states, massive. *)
+		| Prop_Gauge x -> Prop_Col_Gauge x
+		| Prop_Rxi x -> Prop_Col_Rxi x
       | _ -> failwith ("Colorize.It().colorize_propagator: not possible!")
 
     let propagator = function
diff --recursive -u a/src/coupling.mli b/src/coupling.mli
--- a/src/coupling.mli	2009-08-04 17:28:10.000000000 +0200
+++ b/src/coupling.mli	2009-08-04 17:28:22.000000000 +0200
@@ -131,7 +131,7 @@
   | Prop_Unitarity | Prop_Feynman | Prop_Gauge of 'a | Prop_Rxi of 'a
   | Prop_Tensor_2 | Prop_Vectorspinor
   | Prop_Col_Scalar | Prop_Col_Feynman | Prop_Col_Majorana 
-  | Prop_Col_Unitarity 
+  | Prop_Col_Unitarity | Prop_Col_Gauge of 'a | Prop_Col_Rxi of 'a
   | Aux_Scalar | Aux_Vector | Aux_Tensor_1
   | Aux_Spinor | Aux_ConjSpinor | Aux_Majorana
   | Only_Insertion
diff --recursive -u a/src/omega.ml b/src/omega.ml
--- a/src/omega.ml	2009-08-04 17:28:10.000000000 +0200
+++ b/src/omega.ml	2009-08-04 17:28:22.000000000 +0200
@@ -511,8 +511,10 @@
                     | Color.AdjSUN _ -> Some ("gluon", lbl)
                     | _ -> Some ("photon", lbl)
                     end;
+					 | Coupling.Prop_Col_Gauge _ -> Some ("dashes", lbl)
                 | Coupling.Prop_Unitarity
                 | Coupling.Prop_Rxi _ -> Some ("photon", lbl)
+					 | Coupling.Prop_Col_Rxi _ -> Some ("dashes", lbl)
                 | Coupling.Prop_Majorana -> Some ("plain", lbl)
                 | Coupling.Prop_Scalar 
                 | Coupling.Prop_Col_Scalar -> Some ("dashes", lbl)
diff --recursive -u a/src/targets.ml b/src/targets.ml
--- a/src/targets.ml	2009-08-04 17:28:10.000000000 +0200
+++ b/src/targets.ml	2009-08-04 17:28:22.000000000 +0200
@@ -1435,8 +1435,12 @@
       | Prop_Col_Feynman -> 
           printf "(-1.0_omega_prec/3.0_omega_prec) * pr_feynman(%s," p
       | Prop_Gauge xi -> printf "pr_gauge(%s,%s," p (M.gauge_symbol xi)
+      | Prop_Col_Gauge xi ->
+			printf "(-1.0_omega_prec/3.0_omega_prec) * pr_gauge(%s,%s," p (M.gauge_symbol xi)
       | Prop_Rxi xi ->
           printf "pr_rxi(%s,%s,%s,%s," p m w (M.gauge_symbol xi)
+      | Prop_Col_Rxi xi ->
+          printf "(-1.0_omega_prec/3.0_omega_prex) * pr_rxi(%s,%s,%s,%s," p m w (M.gauge_symbol xi)
       | Prop_Tensor_2 -> printf "pr_tensor(%s,%s,%s," p m w
       | Prop_Vectorspinor -> printf "pr_grav(%s,%s,%s,"
             p m w
@@ -1464,8 +1468,8 @@
           printf "(-1.0_omega_prec/3.0_omega_prec) * pj_unitarity(%s,%s,%s," p m gamma
       | Prop_Feynman | Prop_Col_Feynman -> 
           invalid_arg "no on-shell Feynman propagator!" 
-      | Prop_Gauge xi -> invalid_arg "no on-shell massless gauge propagator!"
-      | Prop_Rxi xi -> invalid_arg "no on-shell Rxi propagator!"
+      | Prop_Gauge _ | Prop_Col_Gauge _ -> invalid_arg "no on-shell massless gauge propagator!"
+      | Prop_Rxi _ | Prop_Col_Rxi _ -> invalid_arg "no on-shell Rxi propagator!"
       | Prop_Vectorspinor -> printf "pj_grav(%s,%s,%s," p m gamma
       | Prop_Tensor_2 -> printf "pj_tensor(%s,%s,%s," p m gamma
       | Aux_Scalar | Aux_Spinor | Aux_ConjSpinor | Aux_Majorana
@@ -1488,8 +1492,8 @@
       | Prop_Unitarity -> printf "pg_unitarity(%s,%s,%s," p m gamma
       | Prop_Feynman | Prop_Col_Feynman -> 
           invalid_arg "no on-shell Feynman propagator!" 
-      | Prop_Gauge xi -> invalid_arg "no on-shell massless gauge propagator!"
-      | Prop_Rxi xi -> invalid_arg "no on-shell Rxi propagator!"
+      | Prop_Gauge _ | Prop_Col_Gauge _ -> invalid_arg "no on-shell massless gauge propagator!"
+      | Prop_Rxi _ | Prop_Col_Rxi _ -> invalid_arg "no on-shell Rxi propagator!"
       | Prop_Tensor_2 -> printf "pg_tensor(%s,%s,%s," p m gamma
       | Aux_Scalar | Aux_Spinor | Aux_ConjSpinor | Aux_Majorana
       | Aux_Vector | Aux_Tensor_1 -> printf "("
@@ -4674,8 +4678,11 @@
           failwith "print_fusion: Prop_Col_Feynman not implemented yet!"
       | Prop_Gauge xi ->
           failwith "print_fusion: Prop_Gauge not implemented yet!"
+		| Prop_Col_Gauge _ -> failwith "print_fusion: Prop_Col_Gauge not implemented yet!"
       | Prop_Rxi xi ->
           failwith "print_fusion: Prop_Rxi not implemented yet!"
+		| Prop_Col_Rxi _ ->
+		    failwith "print_fusion: Prop_Col_Rxi not implemented yet!"
       | Prop_Vectorspinor -> 
           failwith "print_fusion: Prop_Vectorspinor not implemented yet!"
       | Prop_Tensor_2 -> propagate ovm_PROPAGATE_TENSOR2
